diff --git a/build/kernel/kleaf/common_kernels.bzl b/build/kernel/kleaf/common_kernels.bzl
--- a/build/kernel/kleaf/common_kernels.bzl
+++ b/build/kernel/kleaf/common_kernels.bzl
@@ -556,7 +556,12 @@ def define_common_kernels(
                 "include/",
                 "kernel/",
                 "tools/",
-            ]),
+            ]) + ([
+                "@//aosp/tools/bpf/resolve_btfids:resolve_btfids_sources",
+                "@//aosp/tools/build:build_tool_sources",
+                "@//aosp/tools/lib/bpf:libbpf_sources",
+                "@//aosp/tools/lib/subcmd:libsubcmd_sources",
+            ] if native.package_name() == "aosp" else []),
             name,
             target_config in target_configs.items():
         native.filegroup(
